src/test/groovy/com/mateusz/jakuszko/roomforyou/service/ApartmentDbServiceSpecification.groovy

```diff
@@ -0,0 +1,142 @@
+        Apartment apartment = createApartment()
+
+        when:
+        def savedApartment = saveAndGetApartmentFromDb(apartment)
+
+        then:
+        notThrown(NotFoundException)
+        checkIfFieldsAreCorrect(savedApartment, apartment)
+    }
+
+    def "should save apartments and return list of apartments after pull from db"() {
+        given:
+        createAndSaveApartments()
+
+        when: "pull apartments from db"
+        List<Apartment> apartments = service.getApartments()
+
+        then: "size o apartments equals 2"
+        apartments.size() == 2
+        and: "assert all fields"
+        apartments.get(0).getId() != null
+        apartments.get(0).getLatitude() == 11
+        apartments.get(0).getLongitude() == 12
+        apartments.get(0).getStreet() == "WallStreet"
+        apartments.get(0).getStreetNumber() == "13"
+        apartments.get(0).getApartmentNumber() == 14
+    }
+
+    def "should update existing apartment"() {
+        given:
+        Apartment apartment = createApartment()
+        Apartment savedApartment = saveAndGetApartmentFromDb(apartment)
+
+        when:
+        Apartment updatedApartment = changeFieldsUpdateAndGetApartmentFromDb(savedApartment)
+
+        then:
+        notThrown(NotFoundException)
+        updatedApartment.getCity() == "Warsaw"
+        updatedApartment.getLatitude() == 23.2
+        updatedApartment.getLongitude() == 52.2
+        updatedApartment.getApartmentNumber() == 23
+        updatedApartment.getStreet() == "Pilsudskiego"
+        updatedApartment.getStreetNumber() == "43"
+    }
+
+    def "should not found an apartment when it hase been deleted"() {
+        given: "create and save apartment into database"
+        Apartment apartment = createApartment()
+        service.save(apartment)
+        Long id = apartment.getId()
+
+        when: "delete apartment and try to pull it from database"
+        service.delete(id)
+        service.getApartment(id).orElseThrow(NotFoundException::new)
+
+        then: "apartment shouldn't found"
+        thrown(NotFoundException)
+    }
+
+    private static Apartment createApartment() {
+        return Apartment.builder()
+                .latitude(11.0)
+                .longitude(12.0)
+                .street("WallStreet")
+                .streetNumber("13")
+                .apartmentNumber(14)
+                .reservations(new ArrayList<Reservation>())
+                .build()
+    }
+
+    private Apartment changeFieldsUpdateAndGetApartmentFromDb(Apartment savedApartment) {
+        savedApartment.setCity("Warsaw")
+        savedApartment.setLatitude(23.2)
+        savedApartment.setLongitude(52.2)
+        savedApartment.setApartmentNumber(23)
+        savedApartment.setStreet("Pilsudskiego")
+        savedApartment.setStreetNumber("43")
+        service.update(savedApartment)
+        Apartment updatedApartment = service.getApartment(savedApartment.getId()).orElseThrow(NotFoundException::new)
+        updatedApartment
```

Comment:
Są różne szkoły, jedni dodają ten "return" na końcu inni nie 🤷‍♂️  Moim zdaniem opcja nr 1 sprawdza się lepiej i pozwala uniknąć wielu pomyłek ale być może to tylko moja preferencja 😄

Reply:
Metoda ostatecznie usunięta, ale zawsze będę dawał return, dzięki za cenne wskazówki